### PR TITLE
Refactor prof log

### PR DESCRIPTION
**Pull request by yinan1048576**
`prof.c` is growing too long, so trying to modularize it.  There are
a few internal functions that had to be exposed but I think it is a
fair trade-off.

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -117,6 +117,7 @@ C_SRCS := $(srcroot)src/jemalloc.c \
 	$(srcroot)src/pages.c \
 	$(srcroot)src/prng.c \
 	$(srcroot)src/prof.c \
+	$(srcroot)src/prof_log.c \
```
Comment (davidtgoldblatt) [inline Makefile.in]: I think the addition of a new source file means that the MSVC project also needs to change.

**Comment by yinan1048576** [comment]
Added the new source file to MSVC project.

**Comment by yinan1048576** [comment]
Rebase.

**Comment by yinan1048576** [comment]
Rebase.

**Comment by yinan1048576** [comment]
Rebase.

**Pull request merged**